controllers/imageclusterinstall_controller.go

```diff
@@ -456,7 +471,18 @@ func (r *ImageClusterInstallReconciler) writeInputData(ctx context.Context, log
 	return ctrl.Result{}, nil
 }
 
-func (r *ImageClusterInstallReconciler) writeClusterInfo(info *clusterinfo.ClusterInfo, file string) error {
+func (r *ImageClusterInstallReconciler) writeClusterInfo(ici *relocationv1alpha1.ImageClusterInstall, cd *hivev1.ClusterDeployment, file string) error {
+	info := clusterinfo.ClusterInfo{
+		Version:         ici.Spec.Version,
+		Domain:          cd.Spec.BaseDomain,
+		ClusterName:     cd.Spec.ClusterName,
+		MasterIP:        ici.Spec.MasterIP,
+		ReleaseRegistry: ici.Spec.ReleaseRegistry,
+		Hostname:        ici.Spec.Hostname,
+	}
+	if ici.Spec.ClusterMetadata != nil {
+		info.ClusterID = ici.Spec.ClusterMetadata.ClusterID
```

Comment:
Using `ClusterMetadata` here is a bit of a problem.

Testing this is giving me this error when trying to create the ICI:
```
The ImageClusterInstall "test-sno" is invalid: 
* spec.clusterMetadata.adminKubeconfigSecretRef: Required value
* spec.clusterMetadata.infraID: Required value
```

If we're requiring the user to provide the cluster ID here they're definitely not going to have values for these other two (and I'm not even sure what we'd use for `infraID`)

Reply:
Let's not use clusterMetadata